impacted-test-engine/src/main/java/com/teamscale/test_impacted/engine/executor/ImpactedTestsProvider.java

```diff
@@ -57,4 +62,24 @@ public ImpactedTestsProvider(TeamscaleClient client, Long baseline, CommitDescri
 		}
 		return null;
 	}
+
+	/**
+	 * Checks that the number of tests returned by Teamscale matches the number of available tests when running with
+	 * {@link #includeNonImpacted}.
+	 */
+	private boolean testCountIsPlausible(List<PrioritizableTestCluster> testClusters,
+										 List<ClusteredTestDetails> availableTestDetails) {
+		if (!this.includeNonImpacted) {
+			return true;
+		}
+		long returnedTests = testClusters.stream().mapToLong(g -> g.tests.size()).sum();
+		if (returnedTests == availableTestDetails.size()) {
+			return true;
+		} else {
+			LOGGER.error(
+					() -> "Retrieved " + returnedTests + " tests from Teamscale, but expected " + availableTestDetails
+							.size() + ".");
```

Comment:
Just wanted to make sure you knew that the following form would also work: `LOGGER.error("Retrieved {} tests from Teamscale, but expected {}.", returnedTests, availableTestDetails.size());`
If you prefer the explicit form, that's okay.

Reply:
Oh wait, I overlooked the lambda

Reply:
So yeah, with lambdas it looks like this: `LOGGER.error("Retrieved {} tests from Teamscale, but expected {}.", () -> returnedTests, () -> availableTestDetails.size());`
Actually I think you can just lose the lambda, since computing this is basically costing you nothing.

Reply:
This is a `org.junit.platform.commons.logging.Logger`, which only has the supplier interface